### PR TITLE
Update group topic

### PR DESCRIPTION
**Pull request by binaryseed**
Extends the `Slack.State` to update the group topic when a change event comes in...

### CODE REVIEW

File: lib/slack/state.ex [hunk 1]
```diff
@@ -53,6 +53,17 @@ defmodule Slack.State do
...
   end
 
+  def update(
+        %{type: "message", subtype: "group_topic", channel: channel, user: user, topic: topic},
+        slack
+      ) do
+    update_in(
+      slack,
+      [:groups, Access.key(channel)],
+      &Map.put(&1, :topic, %{creator: user, last_set: System.system_time(:second), value: topic})
```
Comment (BlakeWilliams) [inline lib/slack/state.ex]: What's the need/idea behind `last_set`? It doesn't look like we're using it, is the thought that the consumer would use it? If so, what's the use case?

File: lib/slack/state.ex [hunk 2]
```diff
@@ -53,6 +53,17 @@ defmodule Slack.State do
     update_in(slack, [:groups, channel, :members], &Enum.uniq([user | &1]))
   end
 
+  def update(
+        %{type: "message", subtype: "group_topic", channel: channel, user: user, topic: topic},
+        slack
+      ) do
+    update_in(
```
Comment (BlakeWilliams) [inline lib/slack/state.ex]: It looks like setting group topic is a an overriding update to a value according to `https://api.slack.com/events/message/group_topic`. If that's the case, could we use `put_in`?

File: lib/slack/state.ex (hunk 1 above)
Comment (binaryseed) [reply lib/slack/state.ex]: It's part of the `topic` map that's there to start, so I just mirrored that.

**Comment by BlakeWilliams** [comment]
thank you!

**Pull request merged**